nano/node/bootstrap/bootstrap_server.cpp

```diff
@@ -25,7 +25,42 @@ void nano::bootstrap_listener::start ()
 		node.logger.always_log (boost::str (boost::format ("Network: Error while binding for incoming TCP/bootstrap on port %1%: %2%") % listening_socket->listening_port () % ec.message ()));
 		throw std::runtime_error (ec.message ());
 	}
-	debug_assert (node.network.endpoint ().port () == listening_socket->listening_port ());
+
+	// the user can either specify a port value in the config or it can leave the choice up to the OS;
+	// independently of user's port choice, he may have also opted to disable UDP or not; this gives us 4 possibilities:
+	// (1): UDP enabled, port specified
+	// (2): UDP enabled, port not specified
+	// (3): UDP disabled, port specified
+	// (4): UDP disabled, port not specified
+	//
+	const auto listening_port = listening_socket->listening_port ();
+	if (!node.flags.disable_udp)
+	{
+		// (1) and (2) -- no matter if (1) or (2), since UDP socket binding happens before this TCP socket binding,
```

Comment:
Where does the udp port binding happens?

Reply:
Bind time/places listed below:

- UDP: `node::node` -> `network::network` -> `udp_channels::udp_channels` -> `udp.cpp:71`
- TCP: `node::start` -> `bootstrap_listener::start` -> `nano::server_socket::start` -> `socket.cpp:210`
-